signalfx-tracing/signalfx-nodejs-tracing/mongo/logger/client.js

```diff
@@ -0,0 +1,77 @@
+// Here we import the tracer before the request library
+// to ensure that its http dependency is instrumented
+const tracer = require('./tracer')
+const config = require('./config')
+
+const readline = require('readline')
+const request = require('request')
+const uuid = require('uuid/v1')
+
+const id = uuid()
+
+function getLog (id) {
+  // obtains the log from the logger server
+  return new Promise((resolve, reject) => {
+    request({ method: 'GET', uri: `${config.serverUrl}/${id}` },
+      (error, response, body) => {
+        if (error) {
+          console.error(error)
+          return reject(error)
+        }
+        let parsed
+        try {
+          parsed = JSON.parse(body)
+        } catch (e) {
+          console.error(e)
+          return reject(e)
+        }
+        if (response.statusCode !== 200) {
+          return resolve(parsed)
+        }
+        resolve(parsed.logs)
+      })
+  })
+}
+
+function logPrompt () {
+  return new Promise((resolve, reject) => {
+    // Initializes the readline session and sends the
+    // logger server all new lines as individual segments
+    const rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout
+    })
+
+    rl.setPrompt('log ~ ')
+
+    let sentLog = false
+    rl.on('close', (e) => {
+      rl.setPrompt('')
+      if (sentLog) {
+        console.log(`Log saved: ${id}`)
+        resolve()
+      }
+      console.log('Goodbye!')
+    })
+
+    rl.on('line', (input) => {
+      if (!input) {
+        return rl.prompt()
+      } else if (input === '/q') {
+        return rl.close()
+      }
+      request({ method: 'POST', uri: config.serverUrl, json: { id, input } },
```

Comment:
Maybe show some manipulation of the auto-instrumented http client span if possible.

Reply:
Not able to intuitively do it w/ request (would need to add response event handlers), so I added parent spans to accomplish this.